### PR TITLE
DEVPROD-1442 Remove stale sonarqube comments on PRs

### PR DESCRIPTION
**Pull request by wejendorp**
The idea is to reduce the noise from sonar when working on long lived PRs via github actions.
Proposed implementation: remove all old comments in the post-action step, leaving only the latest on the PR.

## Result

workflow
pr

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -50,6 +52,9 @@ async function run(): Promise<void> {
 }
 
 async function post(): Promise<void> {
+  if (state.actionStartedAt) {
```
Comment (Tsuesun) [inline src/main.ts:55]: If I understand this correctly, I don't think you actually need this. The comments are posted by the sonarqube server not by the agent so you can remove all sonar comments and then after that the server will post the comments. I don't see any harm with this though.

**Review by Tsuesun — APPROVED** [review]
LGTM, just a note about the time thingy.

File: action.yml [hunk 2]
```diff
@@ -33,6 +33,8 @@ inputs:
   token:
     description: Sonarqube access token
     required: true
+  github-token:
```
Comment (callebjorkell) [inline action.yml:36]: I don't actually think you need to pass this in separately, you should be able to use the one that is passed automatically in the environment (that's what I did for the github-script action too)

Comment (callebjorkell) [reply action.yml:36]: ... so https://docs.github.com/en/actions/reference/authentication-in-a-workflow in other words.

Comment (wejendorp) [reply action.yml:36]: Uuuh... the way I read it, it's specifically for github-script, it's not available in other actions afaict. You just got my hopes up! Damn you @callebjorkell

Comment (callebjorkell) [reply action.yml:36]: Are you sure about that? You also get it passed on just the regular "run" stuff (so without actions)

**Comment by callebjorkell** [comment]
TEST!

**Review by callebjorkell — APPROVED** [review]
It is now absolutely glorious!

**Review by sorenmat — APPROVED** [review]

**Pull request merged**